core/src/io/github/lonamiwebs/klooni/game/State.java

```diff
@@ -0,0 +1,282 @@
+package io.github.lonamiwebs.klooni.game;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.math.MathUtils;
+import com.badlogic.gdx.math.Vector2;
+
+/*
+    State is an object designed to help validating randomly-generated blocks. Every time a block is
+    inserted into the board, the game's state changes. Different permutation of blocks and different
+    position creates new states. Thus to validate a set of generated blocks, we have to check all
+    possible states. Instead of checking them directly on the board, we create a special class for
+    validation.
+
+    This is used in infinite mode.
+ */
+
+public class State {
+
+    // region Members
+
+    private boolean[][] state;
+    private final int cellCount;
+    private int emptySpace = 0;
+    private static int[] unfitBlock = new int[6];
+    private final static int [][] CHECKER = {{0,1,2},{0,2,1},{1,0,2},{1,2,0},{2,0,1},{2,1,0}};
+    private final static Vector2 ORIGIN = new Vector2(0, -1);
```

Comment:
There are a few magic values here which should have an explanation. Why 6 unfit blocks? Why those values for `CHECKER`? And why is the `ORIGIN` pointing downwards one unit?

Reply:
`CHECKER` is all possible order combination to put blocks in board. Since there are 3 blocks generated in one time, there are 6 possible combination, and if all combination are unfit, there must be this one block that makes one combination unfit. We need to track that unfit block, from 6 tested possibilities.

Regarding to `ORIGIN`, it pointing downwards because somehow I cannot compare values in CanPutPiece, so I cannot make exception for origin points.